File: ros2-py-outputs/agr/autonomous_ground_robot/trajectory_tracker_node.py
"""TrajectoryTracker — node implementation. EDIT ME.

Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
(generation-gap pattern): this file is where demo/application logic lives.
The architectural wiring (endpoints, parameters, lifecycle transitions) is in
trajectory_tracker_node_base.py, which is regenerated on every pipeline run.
"""
import rclpy

from .trajectory_tracker_node_base import TrajectoryTrackerBase


class TrajectoryTracker(TrajectoryTrackerBase):
    """trajectory_tracker application logic."""

    def handle_stateSub(self, msg) -> None:
        """/odom_filtered (Odometry).

        TODO: implement message processing logic.
        """

    def handle_planSub(self, msg) -> None:
        """/plan (Path).

        TODO: implement message processing logic.
        """


def main(args=None):
    rclpy.init(args=args)
    node = TrajectoryTracker()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ros2-cpp-outputs/agr/src/trajectory_tracker_node.cpp
// TrajectoryTracker — node implementation. EDIT ME.
//
// Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
// (generation-gap pattern): this file is where demo/application logic lives.
// The architectural wiring is in trajectory_tracker_node_base.hpp/.cpp, regenerated
// on every pipeline run.
#include "autonomous_ground_robot/trajectory_tracker_node_base.hpp"

class TrajectoryTracker : public TrajectoryTrackerBase
{
public:
  using TrajectoryTrackerBase::TrajectoryTrackerBase;

protected:
  void handle_stateSub(const nav_msgs::msg::Odometry & msg) override
  {
    // /odom_filtered — TODO: implement message processing logic.
    (void)msg;
  }
  void handle_planSub(const nav_msgs::msg::Path & msg) override
  {
    // /plan — TODO: implement message processing logic.
    (void)msg;
  }
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<TrajectoryTracker>();
  rclcpp::spin(node->get_node_base_interface());
  rclcpp::shutdown();
  return 0;
}
